compiler/rustc_resolve/src/imports.rs

```diff
@@ -960,7 +983,7 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
                 );
                 self.r.lint_buffer.buffer_lint(
                     PUB_USE_OF_PRIVATE_EXTERN_CRATE,
-                    import.id,
+                    import.expect_id(),
```

Comment:
Is this the same `id` that was added in the match above?  Could we use that variable instead of introducing a panicking call?